e2e/tests-dfx/pull.bash

```diff
@@ -68,13 +70,13 @@ teardown() {
 
     ## 1.2. pull onchain canisters in "app" project
     cd ../app
-    assert_command dfx pull dep1
-    assert_match "Pulling canister ryjl3-tyaaa-aaaaa-aaaba-cai...
+    assert_command_fail dfx pull dep1
+    assert_contains "Pulling canister ryjl3-tyaaa-aaaaa-aaaba-cai...
 Pulling canister rrkah-fqaaa-aaaaa-aaaaq-cai...
 WARN: \`dfx:deps\` metadata not found in canister rrkah-fqaaa-aaaaa-aaaaq-cai."
 
-    assert_command dfx pull # if not specify canister name, all pull type canisters (dep1, dep2) will be pulled
-    assert_match "Pulling canister ryjl3-tyaaa-aaaaa-aaaba-cai...
+    assert_command_fail dfx pull # if not specify canister name, all pull type canisters (dep1, dep2) will be pulled
```

Comment:
Same question here about having the test show the reason for the expected failure